sentry-spring-jakarta/src/main/java/io/sentry/spring/jakarta/checkin/SentryCheckInAdvice.java

```diff
@@ -0,0 +1,78 @@
+package io.sentry.spring.jakarta.checkin;
+
+import com.jakewharton.nopen.annotation.Open;
+import io.sentry.CheckIn;
+import io.sentry.CheckInStatus;
+import io.sentry.DateUtils;
+import io.sentry.IHub;
+import io.sentry.SentryLevel;
+import io.sentry.protocol.SentryId;
+import io.sentry.util.Objects;
+import java.lang.reflect.Method;
+import org.aopalliance.intercept.MethodInterceptor;
+import org.aopalliance.intercept.MethodInvocation;
+import org.jetbrains.annotations.ApiStatus;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+import org.springframework.aop.support.AopUtils;
+import org.springframework.core.annotation.AnnotationUtils;
+import org.springframework.util.ObjectUtils;
+
+/**
+ * Reports execution of every bean method annotated with {@link SentryCheckIn} as a monitor
+ * check-in.
+ */
+@ApiStatus.Internal
+@ApiStatus.Experimental
+@Open
+public class SentryCheckInAdvice implements MethodInterceptor {
+  private final @NotNull IHub hub;
+
+  public SentryCheckInAdvice(final @NotNull IHub hub) {
+    this.hub = Objects.requireNonNull(hub, "hub is required");
+  }
+
+  @Override
+  public Object invoke(final @NotNull MethodInvocation invocation) throws Throwable {
+    final Method mostSpecificMethod =
+        AopUtils.getMostSpecificMethod(invocation.getMethod(), invocation.getThis().getClass());
+
+    @Nullable
+    SentryCheckIn checkInAnnotation =
+        AnnotationUtils.findAnnotation(mostSpecificMethod, SentryCheckIn.class);
+    if (checkInAnnotation == null) {
```

Comment:
If we keep the `SentryCheckIn` Annotation on the type level, then we need to make some changes here:

```
    if (checkInAnnotation == null) {
      checkInAnnotation =
        AnnotationUtils.findAnnotation(
          mostSpecificMethod.getDeclaringClass(), SentryCheckIn.class);
    }

    if (checkInAnnotation == null) {
      return invocation.proceed();
    }
```

also applies to non-jakarta

Reply:
decided not to support that for now